MAJOR operations option regression after version 7.5 even in latest (7.10)

We tried upgrading from 7.5 into 7.10 but experience major regression in the operations override option. 

- 7.10 not acknowledging the operation override
- 7.9 not acknowledging the operation override
- 7.8 acknowledges the operation override, BUT is generating during compilation **duplicate identifiers** for the operations in override
- 7.7 acknowledges the operation override, BUT is generating during compilation **duplicate identifiers** for the operations in override
- 7.6 acknowledges the operation override, BUT is generating during compilation **duplicate identifiers** for the operations in override
- 7.5 the override feature works perfectly, and no duplicate identifiers!

Since version 7.9 and 7.10 we experience a bug where the operations override itself is not working.

We have in orval config (only two operations):

```
      override: {
         ...
        operations: {
          reprint: {
            query: {
              useMutation: true,
            },
          },
          getAddressSuggestions: {
            query: {
              useQuery: true,
            },
          },
        },
```

But this operation id is not changed in a query but stays a mutation. 
The `getAddressSuggestions` is only one time in our open api spec.

### Generated hooks
> you can clearly see they stay mutations for version 7.9 and 7.10

```
export const useReprintMutationOptions = <TError = ErrorType<ErrorBody | void>, TContext = unknown>(options?: {
  mutation?: UseMutationOptions<
```

```
export const useGetAddressSuggestions = <TError = ErrorType<ErrorBody | void>, TContext = unknown>(
  options?: {
    mutation?: UseMutationOptions<
```

It would be nice when https://github.com/orval-labs/orval/pull/2079 had a test to validate this regression/solution? Then we had a real solution instead :)

## 7.8 the override works ...
After this version it stays the original operation instead of the wanted change
So in 7.8 i can see clearly the POST is changed in hook with QUERY and not MUTATION. But with another side effect namely duplicate identifiers.

<img width="423" height="133" alt="Image" src="https://github.com/user-attachments/assets/cbe042f1-e8e1-47bb-b9c6-17c363e2ff0a" />

**but we are getting for stage `Transpiling & type-checking TypeScript` now (also in 7.7 version):**

```
generated/orval/index.ts(1986,17): error TS2300: Duplicate identifier 'useGetAddressSuggestions'.
generated/orval/index.ts(1996,17): error TS2300: Duplicate identifier 'useGetAddressSuggestions'.
generated/orval/index.ts(2006,17): error TS2300: Duplicate identifier 'useGetAddressSuggestions'.
generated/orval/index.ts(2011,17): error TS2300: Duplicate identifier 'useGetAddressSuggestions'.
generated/orval/index.ts(2057,18): error TS2300: Duplicate identifier 'useGetAddressSuggestions'.

generated/orval/index.ts(2434,17): error TS2300: Duplicate identifier 'useReprint'.
generated/orval/index.ts(2446,17): error TS2300: Duplicate identifier 'useReprint'.
generated/orval/index.ts(2458,17): error TS2300: Duplicate identifier 'useReprint'.
generated/orval/index.ts(2465,17): error TS2300: Duplicate identifier 'useReprint'.
generated/orval/index.ts(2513,18): error TS2300: Duplicate identifier 'useReprint'.
```

It all seems to be related to the override option.

--- a/packages/query/src/index.ts
+++ b/packages/query/src/index.ts
@@ -1187,7 +1187,6 @@ const generateQueryHook = async (
     hasOperationQueryOption;
 
   let isMutation = override.query.useMutation && verb !== Verbs.GET;
-  
 
   if (operationQueryOptions?.useMutation !== undefined) {
     isMutation = operationQueryOptions.useMutation;

--- a/packages/query/src/index.ts
+++ b/packages/query/src/index.ts
@@ -1168,23 +1168,26 @@ const generateQueryHook = async (
   let implementation = '';
   let mutators = undefined;
 
+  // Allows operationQueryOptions (which is the Orval config override for the operationId)
+  // to override non-GET verbs
+  const hasOperationQueryOption = !!(
+    operationQueryOptions &&
+    (operationQueryOptions?.useQuery ||
+      operationQueryOptions?.useSuspenseQuery ||
+      operationQueryOptions?.useInfinite ||
+      operationQueryOptions?.useSuspenseInfiniteQuery)
+  );
+
   let isQuery =
-    Verbs.GET === verb &&
-    (override.query.useQuery ||
-      override.query.useSuspenseQuery ||
-      override.query.useInfinite ||
-      override.query.useSuspenseInfiniteQuery);
-
-  // Allows operationQueryOptions to override non-GET verbs
-  const hasOperationQueryOption =
-    operationQueryOptions?.useQuery ||
-    operationQueryOptions?.useSuspenseQuery ||
-    operationQueryOptions?.useInfinite ||
-    operationQueryOptions?.useSuspenseInfiniteQuery;
-
-  isQuery = isQuery || hasOperationQueryOption;
+    (Verbs.GET === verb &&
+      (override.query.useQuery ||
+        override.query.useSuspenseQuery ||
+        override.query.useInfinite ||
+        override.query.useSuspenseInfiniteQuery)) ||
+    hasOperationQueryOption;
 
   let isMutation = override.query.useMutation && verb !== Verbs.GET;
+  
 
   if (operationQueryOptions?.useMutation !== undefined) {
     isMutation = operationQueryOptions.useMutation;

